app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -44,10 +44,16 @@ protected Void doInBackground(Mood ... moods ) {
                     }
                     else{
                         Log.i("Error", "Elasticsearch was not able to add the mood");
+                        mood.setAdd();
+                        UpdateQueueController updateQueueController = FeelTripApplication.getUpdateQueueController();
```

Comment:
If elasticsearch update fails for whatever reason (i.e. no internet), we push the desired tweet to a local queue. This queue will be searched once the condition is resolved (i.e. gain internet) and effectively pushed to elastic search.